uplink/metainfo/batch.go

```diff
@@ -0,0 +1,327 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package metainfo
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+)
+
+var (
+	// ErrInvalidType error for inalid response type casting
+	ErrInvalidType = errs.New("invalid response type")
+)
+
+// Batch represents sending requests in batch
+type Batch struct {
+	client   pb.MetainfoClient
+	requests []*pb.BatchRequestItem
+}
+
+// AddCreateBucket adds request to batch
+func (batch *Batch) AddCreateBucket(params CreateBucketParams) {
```

Comment:
I would flip the assignments and accept an []Request in the batch client call, by making a method like:

```
func (params *CreateBucketParams) BatchItem() *pb.BatchRequestItem {
```

This would allow to define an interface `type BatchItem interface { BatchItem() *pb.BatchRequestItem }`. Then client would accept an argument called `...BatchItem`.

Reply:
I like it, thanks!

I guess with one PR in future it would be good to rename `*Params` types into `*Request` to be more consistent.